os/kernel/task/task_restart.c

```diff
@@ -164,6 +167,18 @@ int task_restart(pid_t pid)
 			return ERROR;
 		}
 
+#if defined(CONFIG_SCHED_ATEXIT) || defined(CONFIG_SCHED_ONEXIT)
+		/* If exit function(s) were registered, call them now before we do any un-
+		 * initialization.
+		 */
+
+		task_atexit((FAR struct tcb_s *)tcb);
```

Comment:
tcb is defined as struct tcb_s *. I think we don't need to type cast.

Reply:
no. it is  task_tcb_s type. so casting is needed.